core/templates/dev/head/pages/exploration-player-page/services/exploration-recommendations.service.ts

```diff
@@ -44,8 +44,9 @@ angular.module('oppia').factory('ExplorationRecommendationsService', [
           collection_id: null,
           include_system_recommendations: null
         };
-        if (GLOBALS.collectionId) {
-          recommendationsUrlParams.collection_id = GLOBALS.collectionId;
+        var collectionId = UrlService.getCollectionIdFromExplorationUrl();
+        if (collectionId) {
```

Comment:
You don't need the if here again just do `collection_id: UrlService.getCollectionIdFromExplorationUrl()` on line 44.